### PR TITLE
@starsirius => Fix occasionally empty primary carousel

### LINKED ISSUE #95: /galleries primary carousel occasionally appears empty
Currently the primary carousel on /galleries appears empty for a couple of our cities, creating an awkward blank space between the filters and the rails below.

Normally [the criteria for selecting carousel shows](https://github.com/artsy/force/blob/master/apps/galleries_institutions/components/primary_carousel/fetch.coffee#L26) for featured cities yields at least one result, but that is not the case right now for Miami and Milan

Perhaps we are seeing this only now bc of the low number of [`current_displayable_shows`](https://github.com/artsy/gravity/blob/master/app/models/util/subscription_rule_ui_helper.rb#L14) in August.

We could either hide the carousel if it's empty, or have fallback criteria to ensure that is populated with >0 shows.

gpc

### PR DESCRIPTION
**Pull request by anandaroop**
Fixes #95 

Occasionally the filter pulldown selections on /gallleries will result in an empty carousel, and a big blank space on the page, because there are no eligible profiles to display in it.

Let's selectively only render the carousel based on if we _actually_ get any profiles back for display. (Watch for Miami in the screencap below)

empty

### CODE REVIEW

File: desktop/apps/galleries_institutions/components/primary_carousel/view.coffee [hunk 1]
```diff
@@ -22,7 +22,11 @@ module.exports = class PrimaryCarousel extends Backbone.View
   fetch: (params) ->
     promise = fetchProfiles(params.toJSON())
     promise.then (profiles) =>
-      @profiles.reset profiles.models
+      if profiles?.models?.length
+        @$el.fadeIn 500
+        @profiles.reset profiles.models
+      else
+        @$el.hide()
```
Comment (starsirius) [inline desktop/apps/galleries_institutions/components/primary_carousel/view.coffee]: An alternative may be to do a `length` check in the [template](https://github.com/artsy/force/blob/aac0b547c620336e42a211c80a53d35b052a862b/desktop/apps/galleries_institutions/components/primary_carousel/template.jade#L8), so we don't need to show/hide additionally. But I could be wrong here.

Comment (anandaroop) [inline desktop/apps/galleries_institutions/components/primary_carousel/view.coffee]: Good idea, I think that will simplify testing drastically.

File: desktop/apps/galleries_institutions/components/primary_carousel/template.jade [hunk 2]
```diff
@@ -1,52 +1,54 @@
 - var length = profiles.length
 
-if length > 1
-  .gpc-bumpers.js-gpc-navigation
-    .gpc-bumper-prev.gi-carousel-arrow-prev.js-gpc-prev: i.icon-chevron-left
-    .gpc-bumber-next.gi-carousel-arrow-next.js-gpc-next: i.icon-chevron-right
+if length > 0
```
Comment (anandaroop) [inline desktop/apps/galleries_institutions/components/primary_carousel/template.jade]: Only render the carousel at all if we have profiles to show. This one line is the main change, everything else below is just whitespace changes from adding that `if`

File: desktop/apps/galleries_institutions/components/primary_carousel/test/template.coffee [hunk 3]
```diff
@@ -6,18 +6,36 @@ Profile = require '../../../../../models/profile'
...
+    describe 'with more than one profile to show', ->    
+      it 'shows navigation arrows', ->
+        @html = template profiles: [@profile, @profile]
+        @$ = $.load @html
+        @$*('.gpc-bumpers').should.not.be.empty()
+
+  describe 'with no profiles to show', ->
+    it 'does not display the carousel', ->
+      @html = template profiles: []
+      @html.should.be.empty()
```
Comment (anandaroop) [inline desktop/apps/galleries_institutions/components/primary_carousel/test/template.coffee]: Ahh, such a simple test 😎. 

I also added a few specs above to make the behavior of the template clearer for all the difference cases.

File: desktop/apps/galleries_institutions/components/primary_carousel/template.jade (hunk 2 above)
Comment (starsirius) [inline desktop/apps/galleries_institutions/components/primary_carousel/template.jade]: 👍

File: desktop/apps/galleries_institutions/components/primary_carousel/test/template.coffee (hunk 3 above)
Comment (starsirius) [inline desktop/apps/galleries_institutions/components/primary_carousel/test/template.coffee]: Sooooo simple! 💯

**Comment by starsirius** [comment]
Merging! Thanks for the fix, @anandaroop!!

**Pull request merged**